src/backend/functions/projects-edit.ts

```diff
@@ -0,0 +1,492 @@
+  );
+  const goalsChangeJson = createDescriptionBulletChangesJson(
+    originalProject.goals.map((element) => descBulletConverter(element)),
+    goals,
+    crId,
+    userId,
+    wbsElementId,
+    'goals'
+  );
+  const featuresChangeJson = createDescriptionBulletChangesJson(
+    originalProject.features.map((element) => descBulletConverter(element)),
+    features,
+    crId,
+    userId,
+    wbsElementId,
+    'features'
+  );
+  const otherConstraintsChangeJson = createDescriptionBulletChangesJson(
+    originalProject.otherConstraints.map((element) => descBulletConverter(element)),
+    otherConstraints,
+    crId,
+    userId,
+    wbsElementId,
+    'other constraints'
+  );
+  // add the changes for each of dependencies, expected activities, and deliverables
+  changes = changes
+    .concat(rulesChangeJson)
+    .concat(goalsChangeJson.changes)
+    .concat(featuresChangeJson.changes)
+    .concat(otherConstraintsChangeJson.changes);
+
+  // update the project with the input fields
+  const updatedProject = await prisma.project.update({
+    where: {
+      wbsElementId
+    },
+    data: {
+      budget,
+      summary,
+      googleDriveFolderLink: body.googleDriveFolderLink,
+      slideDeckLink: body.slideDeckLink,
+      bomLink: body.bomLink,
+      taskListLink: body.taskListLink,
+      rules,
+      wbsElement: {
+        update: {
+          name,
+          status: wbsElementStatus,
+          projectLeadId: body.projectLead,
+          projectManagerId: body.projectManager
+        }
+      }
+    }
+  });
+
+  // --> Pointer
+
+  // Update any deleted description bullets to have their date deleted as right now
+  const deletedIds = goalsChangeJson.deletedIds
+    .concat(featuresChangeJson.deletedIds)
+    .concat(otherConstraintsChangeJson.deletedIds);
+  if (deletedIds.length > 0) {
+    await prisma.description_Bullet.updateMany({
+      where: {
+        descriptionId: {
+          in: deletedIds
+        }
+      },
+      data: {
+        dateDeleted: new Date()
+      }
+    });
+  }
+  addDescriptionBullets(goalsChangeJson.addedDetails, updatedProject.projectId, 'projectIdGoals');
+  addDescriptionBullets(
+    featuresChangeJson.addedDetails,
+    updatedProject.projectId,
+    'projectIdFeatures'
+  );
+  addDescriptionBullets(
+    otherConstraintsChangeJson.addedDetails,
+    updatedProject.projectId,
+    'projectIdOtherConstraints'
+  );
+  editDescriptionBullets(
+    goalsChangeJson.editedIdsAndDetails
+      .concat(featuresChangeJson.editedIdsAndDetails)
+      .concat(otherConstraintsChangeJson.editedIdsAndDetails)
+  );
+  // create the changes in prisma
+  await prisma.change.createMany({
+    data: changes
+  });
+
+  // return the updated work package
+  return buildSuccessResponse(updatedProject);
+};
+
+// /**
+//  * HELPER METHODS:
+//  */
+
+// helper method to add the given description bullets into the database, linked to the given work package
+const addDescriptionBullets = async (
```

Comment:
Here, when the description bullets are added to the database, we do not use any id passed (only the id for the relation to the project)